### PR TITLE
add move-item command

### PR DESCRIPTION
**Pull request by noaco**

### CODE REVIEW

**Comment by orius123** [comment]
Should it replace the current `Integrations/integration-Office_EWS.yml` if it should, should we delete the other file?

File: Integrations/integration-Office_EWS.yml [hunk 1]
```diff
@@ -451,9 +455,45 @@ script:
...
+                            '%item-ids%' +
+                        '</m:ItemIds>' +
+                    '</m:MoveItem>' +
+                '</soap:Body>' +
+            '</soap:Envelope>',
+            innerPath: 'Envelope.Body.MoveItemResponse.ResponseMessages.MoveItemResponseMessage.Items.Message',
+            resStatus: 'Envelope.Body.MoveItemResponse.ResponseMessages.MoveItemResponseMessage.-ResponseClass',
+            contextKey: 'EWS.MovedItems',
+            translator: [
+                { to: 'New-Item-Id', from: 'ItemId.-Id'}
```
Comment (asafshen) [inline Integrations/integration-Office_EWS.yml]: does this key only goes to war-room? if so - I think `New item ID` looks a bit better

File: Integrations/integration-Office_EWS.yml [hunk 2]
```diff
@@ -451,9 +455,45 @@ script:
...
+                        '</m:ToFolderId>' +
+                        '<m:ItemIds>' +
+                            '%item-ids%' +
+                        '</m:ItemIds>' +
+                    '</m:MoveItem>' +
+                '</soap:Body>' +
+            '</soap:Envelope>',
+            innerPath: 'Envelope.Body.MoveItemResponse.ResponseMessages.MoveItemResponseMessage.Items.Message',
+            resStatus: 'Envelope.Body.MoveItemResponse.ResponseMessages.MoveItemResponseMessage.-ResponseClass',
+            contextKey: 'EWS.MovedItems',
```
Comment (asafshen) [inline Integrations/integration-Office_EWS.yml]: Regarding context - Optimal result is to also update exiting context object as well
('EWS.Items' & 'EWS.SearchItems' with existing ID)

@aviadl what do you think? we can skip this and if user needs it - he/she can search again

File: Integrations/integration-Office_EWS.yml [hunk 3]
```diff
@@ -1404,5 +1443,24 @@ script:
...
+    arguments:
+    - name: target-mailbox
+      description: The user email address
+    - name: item-ids
+      required: true
+      description: Comma separated list of Item IDs (result of the ews-search-mailbox
+        command)
+    - name: folder-identifier
+      required: true
+      description: The folder id
```
Comment (asafshen) [inline Integrations/integration-Office_EWS.yml]: Please give example also. `(E.g. 'inbox', 'sentitems'). All folder-ids are listed in ews-find-folders command`

**Comment by asafshen** [comment]
image

You are getting this error because `releaseNotes` is not indented to the start of the line
(just remove prefix tab)

File: Integrations/integration-Office_EWS.yml [hunk 4]
```diff
@@ -3,7 +3,6 @@ commonfields:
   version: -1
 name: EWS
 display: EWS
-fromversion: 2.6.0
```
Comment (asafshen) [inline Integrations/integration-Office_EWS.yml]: revert this line please

File: Integrations/integration-Office_EWS.yml [hunk 5]
```diff
@@ -1404,4 +1488,26 @@ script:
...
+        (e.g.'AAMkAGY3OTQyMzMzLW'). All folder-ids are listed in ews-find-folders
+        command.
+    - name: version
+      description: Exchange version (by default 2010, change to 2013 if using 2013).
+        This can also be set in the settings/integrations page
+    outputs:
+    - contextPath: EWS.MovedItems.ItemNewId
+      description: The new item id
+    description: Move one or more item to one destination folder. Note that after
+      an item is moved, a new id will be associated with it.
```
Comment (asafshen) [inline Integrations/integration-Office_EWS.yml]: please add ItemOldID to outputs

File: Integrations/integration-Office_EWS.yml [hunk 6]
```diff
@@ -617,41 +717,26 @@ script:
...
-                    }
                     var disp = buildDisplay(currentCommand, translated);
                     entry.HumanReadable = tableToMarkdown(
                         hrTitle,
                         disp
                     );
+                    translated = currentCommand.dontAddTranslatedToContext ? {} : translated;
+                    entry.EntryContext[currentCommand.contextKey] = createContext(translated, additionalContextFromArgs);
+                    if (currentCommand.addUpdatesToContext) {currentCommand.addUpdatesToContext(entry , additionalContextFromArgs);}
+                    log(JSON.stringify(entry.EntryContext));
```
Comment (asafshen) [inline Integrations/integration-Office_EWS.yml]: please remove logs (others if any)

File: Integrations/integration-Office_EWS.yml [hunk 7]
```diff
@@ -447,17 +476,80 @@ script:
...
+            saveArgsToContext: [
+                {
+                    contextKey:'OldItemId',
+                    argName: 'item-id'
+                }
+            ],
+            addUpdatesToContext: function(entry, savedArgs){
+                var oldID = savedArgs['OldItemId'];
+                var newID = entry.EntryContext['EWS.MovedItems']['NewItemId'];
+                var inContext = function(path) {
```
Comment (asafshen) [inline Integrations/integration-Office_EWS.yml]: please add this function to `script-CommonServer.yml`

**Comment by anara123** [comment]
@asafshen can we approve this PR or still some changes need to be done?

File: Integrations/integration-Office_EWS.yml [hunk 8]
```diff
@@ -447,17 +476,80 @@ script:
...
+            saveArgsToContext: [
+                {
+                    contextKey:'OldItemId',
+                    argName: 'item-id'
+                }
+            ],
+            addUpdatesToContext: function(entry, savedArgs){
+                var oldID = savedArgs['OldItemId'];
+                var newItem = { ID: entry.EntryContext['EWS.MovedItems']['NewItemId']};
+                var inContext = function(path) {
```
Comment (asafshen) [inline Integrations/integration-Office_EWS.yml]: Please move `inContext` to CommonServer

**Comment by dorsha** [comment]
@asafshen / @noaco what is the status of this PR?

**Comment by asafshen** [comment]
@dorsha 
@noaco will resolve conflicts & merge today

**Comment by dorsha** [comment]
?

**Pull request closed without merging**